src/http-client.js

```diff
@@ -93,13 +110,36 @@ export class HttpClient {
       transformers[i](this, processor, message);
     }
 
-    promise = processor.process(this, message).then(response => {
-      trackRequestEnd(this, processor);
-      return response;
-    }).catch(response => {
-      trackRequestEnd(this, processor);
-      throw response;
-    });
+    processRequest = (message) => {
```

Comment:
I think the problem we're going to run into here is that all transformers run inside the processor. Most of the client's API uses transformers to set request properties, so the message that the `request` and `requestError` handlers see will pretty much always be an empty message.

What do you think about breaking up the `processor.process` method so that the message is fully constructed by one method and then sent by another? That would let us apply the transformers & everything to build the complete message, then run the request interceptors, then send the message. I think it would make interceptors a lot more useful since they'd be able to inspect and modify the message with the data that will be sent. It would also let the `requestError` interceptors catch any errors caused by transformers.